app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  has_many :likes
+  has_many :comments
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+
+  def update_post_counter
+    user.update(posts_count: user.posts.count)
+  end
```

Comment:
- [ ] Kindly note that the `user` here is defined through the `author`, so you should be calling `author` instead of calling `user`. Kindly change that 

```suggestion
  belongs_to :author, class_name: 'User', foreign_key: 'author_id'

  def update_post_counter
   author.update(posts_count: user.posts.count)
  end
```